scripts/backup/backup

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+UMBREL_ROOT="$(readlink -f $(dirname "${BASH_SOURCE[0]}")/../..)"
+BACKUP_FOLDER="backup"
+BACKUP_ROOT="${UMBREL_ROOT}/${BACKUP_FOLDER}"
+BACKUP_FILE="${UMBREL_ROOT}/backup.tar.gz.pgp"
+
+check_dependencies () {
+  for cmd in "$@"; do
+    if ! command -v "$cmd" >/dev/null 2>&1; then
+      echo "This script requires \"${cmd}\" to be installed"
+      exit 1
+    fi
+  done
+}
+
+check_dependencies openssl tar gpg shuf curl
+
+# Deterministically derives 128 bits of cryptographically secure entropy
+derive_entropy () {
+  identifier="${1}"
+  umbrel_seed=$(cat "${UMBREL_ROOT}/db/umbrel-seed/seed") || true
+
+  if [[ -z "$umbrel_seed" ]] || [[ -z "$identifier" ]]; then
+    >&2 echo "Missing derivation parameter, this is unsafe, exiting."
+    rm -f "${UMBREL_ROOT}/statuses/backup-in-progress"
+    exit 1
+  fi
+
+  # We need `sed 's/^.* //'` to trim the "(stdin)= " prefix from some versions of openssl
+  echo -n "${identifier}" | openssl dgst -sha256 -hmac "${umbrel_seed}" | sed 's/^.* //'
```

Comment:
Why seed as hmac and identifier as message?

Reply:
You mean why those input parameters or why are they that way round?

Reply:
Why are they that way round

Reply:
I guess because the official parameters are `secret` and `message`. The identifier is not secret, it's just a text string that's unique to that use case to result in a unique entropy. It's not really a "message" but it's closer to being a message than a secret.

The seed is definitely not a message, and it is secret, so seemed more sensible to set it as the `secret` parameter.

But it's an HMAC not just a hash, so it's resistant to length extension attacks, so really I think either way would work just as well.

Had to pick one way and this seemed like the most logical to me.

Do you think it makes more sense the other way?